agent-control/src/bin/main_onhost.rs

```diff
@@ -16,8 +15,38 @@ use std::error::Error;
 use std::process::ExitCode;
 use tracing::{error, info, trace};
 
+#[cfg(target_os = "windows")]
+use newrelic_agent_control::command::windows::{WINDOWS_SERVICE_NAME, setup_windows_service};
+
+#[cfg(target_os = "windows")]
+windows_service::define_windows_service!(ffi_service_main, service_main);
+
 fn main() -> ExitCode {
-    Command::run(AGENT_CONTROL_MODE_ON_HOST, _main)
+    #[cfg(not(target_os = "windows"))]
+    {
+        Command::run(AGENT_CONTROL_MODE_ON_HOST, _main)
+    }
+
+    #[cfg(target_os = "windows")]
+    {
+        if windows_service::service_dispatcher::start(WINDOWS_SERVICE_NAME, ffi_service_main)
+            .is_err()
+        {
+            // Not running as Windows Service, run normally
+            return Command::run(AGENT_CONTROL_MODE_ON_HOST, |cfg, tracer| {
+                _main(cfg, tracer, false)
+            });
+        }
+        ExitCode::SUCCESS
+    }
+}
```

Comment:
Nit: Remove some indentation and conditional checks.
Alternative: Put the imports inside the windows part of `main` to remove some conditional checks.

This is purely subjective. But I find that the more conditional checks that I see the more "noise" there is.

```suggestion
#[cfg(not(target_os = "windows"))]
fn main() -> ExitCode {
    Command::run(AGENT_CONTROL_MODE_ON_HOST, _main)
}

#[cfg(target_os = "windows")]
fn main() -> ExitCode {
    use newrelic_agent_control::command::windows::{WINDOWS_SERVICE_NAME, setup_windows_service};
    windows_service::define_windows_service!(ffi_service_main, service_main);

    if windows_service::service_dispatcher::start(WINDOWS_SERVICE_NAME, ffi_service_main).is_err() {
        // Not running as Windows Service, run normally
        return Command::run(AGENT_CONTROL_MODE_ON_HOST, |cfg, tracer| {
            _main(cfg, tracer, false)
        });
    }
    ExitCode::SUCCESS
}
```